src/components/TransactionHistoryRows/TransactionDetails/NftId.tsx

```diff
@@ -0,0 +1,38 @@
+import type { FlexProps } from '@chakra-ui/react'
+import { Button, Flex, Link } from '@chakra-ui/react'
+
+const Id = ({ id, ...rest }: FlexProps & { id: string }) => (
+  <Flex alignItems='center' whiteSpace='nowrap' {...rest}>
+    <span style={{ lineHeight: 1 }}>
+      {id.length >= 12 ? `${id.slice(0, 4)}...${id.slice(-4)}` : id}
+    </span>
+  </Flex>
+)
+
+export const NftId = ({
+  explorer,
+  id,
+  token,
+}: {
+  explorer?: string
+  id: string
+  token: string
+}) => (
+  <Button
+    as={Link}
+    href={`${explorer}/token/${token}?a=${id}`}
```

Comment:
we can potentially lift this as another asset field `explorerTokenUrl` or something like that. the extra query string makes it a bit weird and all evm chains are just etherscan clones so this worked for now. let me know if anyone has strong opinions here

Reply:
Yeah I'm fine with this since all Etherscan-like behave the same, I don't see the point of adding a new field just for the sake of avoiding recreating the URL from the base URL.
If we want to keep things DRY in the future, can always make a simple one-liner to achieve this